spec/services/container_deployment_service_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'recursive-open-struct'
+
+describe ContainerDeploymentService do
+  before do
+    %w(amazon openstack google azure redhat vmware).each do |p|
+      network = FactoryGirl.create(:network, :ipaddress => "127.0.0.1")
+      @hardware = FactoryGirl.create(:hardware)
+      @hardware.networks << network
+      if (p.include? "amazon") || (p.include? "redhat")
+        template = FactoryGirl.create("template_#{p}".to_sym)
+        provider = FactoryGirl.create("ems_#{p}".to_sym, :miq_templates => [template])
+      else
+        provider = FactoryGirl.create("ems_#{p}".to_sym)
+      end
+      FactoryGirl.create("vm_#{p}".to_sym, :ext_management_system => provider, :hardware => @hardware)
```

Comment:
hmm, this does use the current iteration's `@hardware`.  and you don't use `@hardware` elsewhere, right?